### PR TITLE
Convex hull

### PR DESCRIPTION
**Pull request by JaapWijnen**
This was requested in #202

### CODE REVIEW

**Comment by vincentngo** [comment]
@JaapWijnen looking at this one now.

File: Convex Hull/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Convex Hull
+
+There are multiple Convex Hull algorithms. This particular implementation uses the Quickhull algorithm.
+
+Given a group of points on a plane. The Convex Hull algorithm calculates the shape (made up from the points itself) containing all these points. It can also be used on a collection of points of different dimention. This implementation however covers points on a plane. It essentially calculates the lines between points which together contain all points.
```
Comment (vincentngo) [inline Convex Hull/README.md]: Would it be better to call it an algorithm that generates a polygon?

File: Convex Hull/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+We now define the following recursive function:
+
+`findHull(points: [CGPoint], p1: CGPoint, p2: CGPoint)`
+
+```
+findHull(S1, A, B)
+findHull(S2, B, A)
+```
+
+What this function does is the following:
```
Comment (vincentngo) [inline Convex Hull/README.md]: The bullet points is showing as a big paragraph, instead of line by line

File: Convex Hull/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+1. If `points` is empty we return as there are no points to the right of our line to add to our hull.
+2. Draw a line from `p1` to `p2`.
+3. Find the point in `points` that is furthest away from this line. (`maxPoint`)
+4. Add `maxPoint` to the hull right after `p1`.
+5. Draw a line (`line1`) from `p1` to `maxPoint`.
+6. Draw a line (`line2`) from `maxPoint` to `p2`. (These lines now form a triangle)
+7. All points within this triangle are of course not part of the hull and thus can be ignored. We check which points in `points` are to the right of `line1` these are grouped in an array `s1`.
+8. All points that are to the right of `line2` are grouped in an array `s2`. Note that there are no points that are both to the right of `line1` and `line2` as then `maxPoint` wouldn't be the point furthest away from our initial line between `p1` and `p2`.
+9. We call `findHull(_, _, _)` again on our new groups of points to find more hull points.
+```
```
Comment (vincentngo) [inline Convex Hull/README.md]: Would be nice to have step by step diagrams, or an animation to show how the algorithm works. It took me a while to understand how the algorithm works.

File: Convex Hull/README.md (hunk 1 above)
Comment (tkrajacic) [inline Convex Hull/README.md]: `dimention` -> `dimension`

**Comment by vincentngo** [comment]
@JaapWijnen i'm going to merge this for now, i think the algorithm is really cool. So Whenever you get time or someone else does, we can beef up the explanation on this 👍

**Comment by JaapWijnen** [comment]
Will do! Sorry didn't have a lot of time lately. Not sure how this works. Should I make a new PR or continue working on this branch?

**Comment by vincentngo** [comment]
@JaapWijnen yea new PR is fine! No worries !

**Pull request merged**